src/main/java/com/josdem/vetlog/service/impl/VaccinationServiceImpl.java

```diff
@@ -36,54 +34,18 @@
 @RequiredArgsConstructor
 public class VaccinationServiceImpl implements VaccinationService {
 
-    private static final String DA2PP = "DA2PP";
-    private static final String DEWORMING = "Deworming";
-    private static final String LEPTOSPIROSIS = "Leptospirosis";
-    private static final String RABIES = "Rabies";
-
     private final VaccinationRepository vaccinationRepository;
 
+    private final Map<PetType, VaccinationStrategy> vaccinationStrategies;
+
     @Override
     public void save(Pet pet) {
-        if (pet.getBreed().getType() != PetType.DOG) {
-            throw new BusinessException("Only dogs are allowed");
+        VaccinationStrategy strategy = vaccinationStrategies.get(pet.getBreed().getType());
```

Comment:
Instead of  validating null, I suggest to use `Optional<VaccinationStrategy> optional` so that you can do:
```java
if (options.isEmpty()) {
           throw new BusinessException("No vaccination strategy found for pet type: "
                    + pet.getBreed().getType());
 }
```

Reply:
Okay, thanks for the feedback, I will make the change and make a PR

Reply:
Thank you for doing suggested changes :+1: